README.md

```diff
@@ -71,6 +71,20 @@ async fn main() -> Result<(), Error> {
         .try_collect()
         .await?;
 
+    // Or handle the pagination manually.
+    let parameters = SearchParameters::empty().and_raw("_count", "10");  // Set page size.
+    let page: Page<Patient> = client.search_paged(PagedSearchMethod::Parameters(parameters)).await?;
```

Comment:
I think you might underestimate the changes necessary to actually make this work like you describe here. Our ideas here should be on a simular level, even though mine obviously is slightly bigger.
But I don't want an inconsistent API, so I would like to go the extra mile to provide the intuitive and nice UX that people in the Rust ecosystem are so used to :D  

Besides that a mini thing: your enum that is given to the search fn is kind of like overloading different methods, so I would suggest just making it 2 separate methods for users' convenience. It will internally of course use the same code.

Reply:
Yes, I expect quite chunk of work there, that's why I kept this changeset small to reduce the impact radius, but this is not an approach you want to take, I understand.
If that's the case, you don't see a way this approach gets in; I wouldn't want to take you longer on this, I'll suffice with what we have now, and rather see how the project evolves to know your vision for it, and maybe PR something else later.

Thank you Flix :slightly_smiling_face:

Reply:
Yes, but certainly thanks for yout effort and perspective.
I hope you will continue opening issues for nuisances and missing features. And if you want, potentially work on some where our ideas align.

I did not get time to think and look through the code, I hope on the weekend.
But I do want to support your usecase and I expect to need to offer some kind of manual search method that gets a URL and returns one page. So that means this PR could actually fit into the niche of what the new API looks like. So it might be this small change that allows your use case and still be similar after the refactor.

Reply:
Thank you, I'm looking forward for the future interactions too, I'll keep following to see where/how we align.
Let me know if this PR needs adjustment to be the first one :slightly_smiling_face: